gnd/src/main/java/com/google/android/gnd/persistence/local/LocalDataStore.java

```diff
@@ -50,29 +53,35 @@
 
   /**
    * Returns a long-lived stream that emits the full set of features for a project on subscribe, and
-   * continues to return the full set each time a feature is added/changed/removed. The full set is
-   * returned rather than deltas for simplicity and to implement a fully reactive UI in which each
-   * update is idempotent.
+   * continues to return the full set each time a feature is added/changed/removed.
    */
   Flowable<ImmutableSet<Feature>> getFeaturesOnceAndStream(Project project);
 
+  /**
+   * Returns a long-lived stream that emits the full set of records for a particular feature and
+   * form on subscribe, and continues to return the full set each time a record is
+   * added/changed/removed.
+   */
+  Flowable<ImmutableList<Record>> getRecordsOnceAndStream(Feature feature, String formId);
+
   /** Returns the feature with the specified UUID from the local data store, if found. */
   Maybe<Feature> getFeature(Project project, String featureId);
 
   /** Returns the record with the specified UUID from the local data store, if found. */
   Maybe<Record> getRecord(Feature feature, String recordId);
 
-  /** Returns the records associated with the specified feature, or an empty list if none found. */
-  Single<ImmutableList<Record>> getRecords(Feature feature);
-
   /**
    * Returns all feature and record mutations in the local mutation queue relating to feature with
    * the specified id.
    */
   Single<ImmutableList<Mutation>> getPendingMutations(String featureId);
 
-  /**
-   * Removes the provided feature and record mutations from the local mutation queue.
-   */
+  /** Removes the provided feature and record mutations from the local mutation queue. */
   Completable removePendingMutations(ImmutableList<Mutation> mutations);
+
+  /** Insert or replace feature in the local data store. */
```

Comment:
The problem here is that mergeFeature() is only partly implemented; once we add the ability to edit features (move point or edit shapes) this will be named appropriately. I'll update the Javadoc to reflect this.